specs/sql/README.md

```diff
@@ -335,18 +335,133 @@ contain only column names. An error is raised if more than one
+place of a column name. If a table contains a user defined column named
+`"rowid"`, `"oid"` or `"_rowid_"`, then that name always refers the
+explicitly declared column and cannot be used to retrieve the integer
+`ROWID` value.
+
+The data for Tableland tables are stored in sorted order, by `ROWID`.
+This means that retrieving or sorting records by `ROWID` is fast.
+Searching for a record with a specific `ROWID`, or for all records with
+`ROWID`s within a specified range is around twice as fast as a similar
+search made by specifying any other `PRIMARY KEY`. This `ROWID` sorting
+is also required for sub-queries and other SQL features on Tableland, to
+ensure deterministic ordering of results.
+
+With one exception noted below, if a table has a primary key that
+consists of a single column and the declared type of that column is
+`INTEGER`, then the column becomes an alias for the `ROWID`. Such a
+column is usually referred to as an "integer primary key". A
+`PRIMARY KEY` column only becomes an integer primary key if the declared
+type name is exactly `INTEGER`. Other integer type names like `INT`
+causes the primary key column to behave as an ordinary table column with
+integer affinity and a unique index, not as an alias for the `ROWID`.
+
+In the above case of an integer primary key, there is an additional
+implied `AUTOINCREMENT` constraint, which forces the integer primary key
+to behave as if it were specified with
+`INTEGER PRIMARY KEY AUTOINCREMENT`. See [Autoincrement](#autoincrement)
+for further details.
+
+The exception mentioned above is that if the declaration of a column
+with declared type `INTEGER` includes an `PRIMARY KEY DESC` clause, it
+does not become an alias for the `ROWID` and is not classified as an
+integer primary key. In pratice, this means that the following three
+table declarations all cause the column "x" to be an alias for the
+`ROWID` (an integer primary key):
+
+-   `CREATE TABLE t(x INTEGER PRIMARY KEY ASC, y, z);`
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x ASC));`
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x DESC));`
+
+But the following declaration does not result in "x" being an alias for
+the `ROWID`:
+
+-   `CREATE TABLE t(x INTEGER PRIMARY KEY DESC, y, z);`
+
+Given this, and the implied autoincrement behavior, if the primary key
+is defined at the end as a table constraint and not as a column
+constraint:
+
+-   `CREATE TABLE t(x INTEGER, y, z, PRIMARY KEY(x DESC));`
+
+this will actually be transformed (behind the scenes) by the Tableland
+Parser to:
+
+-   `CREATE TABLE t(x INTEGER PRIMARY KEY DESC AUTOINCREMENT, y, z);`
+
+in order to maintain the `ROWID` alias behavior. This transformation to
+the more canonical direct constraint on the primary key is required to
+enfore the implied `AUTOINCREMENT` behavior on the special integer
+primary keys.
+
+Rowid values *may not* be modified using an `UPDATE` statement in the
```

Comment:
Just to confirm something related to `by using an alias created by an integer primary key`. Given the following example: `create table a (a INTEGER PRIMARY KEY, b TEXT);`

Column `a` is an alias to `rowid`, right? We are blocking any `update` (or `insert`) on `rowid` but I'm afraid we can't block updates on `a`. For example, 

`insert into a VALUES (2, 'Hello');` or `update a set a = 10 where b = 'Hello';`

should work, right?

I think that's fine. The worst-case scenario is somebody updating `a` to that huge number and getting an error on the next insert.

Reply:
Yes, I think we talked about warning people that if you insert some big number X, then you're risking locking your table.
Since we force `AUTOINCREMENT` in all `PRIMARY KEY` cases, that's fine since we won't get into the randomness situation. (This last sentence is mentioned for completeness).